tools/torch_quant/bert_ptq_demo.py

```diff
@@ -41,27 +47,90 @@ def calibrate(model, dataloader, device):
     for batch in dataloader:
         batch = {k: v.to(device) for k, v in batch.items()}
         with torch.no_grad():
-            model(**batch)
+            model(
+                batch["input_ids"],
+                batch["attention_mask"],
+                batch["token_type_ids"],
+            )
 
 
 def evaluate(model, dataloader, device):
-    metric = load_metric("glue", "mrpc")
+    metric = load("glue", "mrpc")
     n_steps = len(dataloader)
     progress_bar = tqdm(total=n_steps, desc='batches')
     model.eval().to(device)
     for batch in dataloader:
         batch = {k: v.to(device) for k, v in batch.items()}
         with torch.no_grad():
-            outputs = model(**batch)
-        predictions = torch.argmax(outputs["logits"], dim=-1)
+            outputs = model(
+                batch["input_ids"],
+                batch["attention_mask"],
+                batch["token_type_ids"],
+            )
+        predictions = torch.argmax(outputs[0], dim=-1)
         metric.add_batch(predictions=predictions, references=batch["labels"])
         progress_bar.update(1)
     return metric.compute()
 
 
+def gen_dummy_inputs(device, batch_size=1, seq_len=128):
+    input_ids = torch.randint(
+        100, 1000, (batch_size, seq_len), dtype=torch.int64, device=device)
+    attention_mask = torch.ones_like(input_ids)
+    token_type_ids = torch.zeros_like(input_ids)
+    return (input_ids, attention_mask, token_type_ids)
+
+
+def benchmark(model, device, n_warmup=10, n_repeat=20):
+    inputs = gen_dummy_inputs(device)
+    for _ in range(n_warmup):
+        with torch.no_grad():
+            model(*inputs)
+    start = time.time()
+    for _ in range(n_repeat):
+        with torch.no_grad():
+            model(*inputs)
+    end = time.time()
+    print(f'benchmark: avg={(end-start)/n_repeat*1000:.2f}ms')
+
+
+def ptq(calib_dataloader, device):
+    print('load model...')
+    model = AutoModelForSequenceClassification.from_pretrained(
+        MODEL_ID, torchscript=True)
+    # model.config.problem_type is set in first forward calling, and is required for fx tracing
+    # we call calibrate() here only as a dummy forward calling.
+    calibrate(model, calib_dataloader, device)
+
+    tracer = HFTracerWrapper(
+        ["input_ids", "attention_mask", "token_type_ids"])
+    quantizer = Quantizer(tracer=tracer, backend=Backend.DISC)
```

Comment:
Is there any interface for us to set the quant module list?

Reply:
There is a `module_filter` argument for `Quantizer` for this purpose. Not implemented yet 😂